docs: add azure native vm based to deployment section (#8985)
mattermost/docs: source/deployment-guide/server/linux/deploy-azure-native-vm.rst

:orphan:
:nosearch:

> 

Author: https://github.com/mattermost/docs/pull/8985/commits/2c08d49bae6dc1c64c64f862b2d9c325bbafc7c2

Mattermost is published as an **Azure Marketplace** solution that provisions a production-style Mattermost stack in your own subscription using Azure-managed services for compute, database, shared file storage, and load balancing. This option is intended for organizations that want a resilient Mattermost deployment they can scale and operate within their existing Azure tenant.

Prerequisites
-------------

> * An active **Azure subscription** with permission to create resource groups and accept **Azure Marketplace** billing terms for the offer.

Reviewer: Can we name the actual Azure RBAC roles needed ?

Author: That's a good point, maybe `An active **Azure subscription** with the **Owner** or **Contributor** role on the subscription (required to purchase and deploy Azure Marketplace solutions), permission to create resource groups, and permission to accept **Azure Marketplace** billing terms for the offer.`

Author: https://github.com/mattermost/docs/pull/8985/commits/c3abdc6b22fa4c98c9c8f564adeddca8d17108b4

2. **Application Gateway with public IP:** Acts as the entry point for Mattermost traffic, with optional **HTTPS listener** (using your PFX certificate) and HTTP-to-HTTPS redirection when HTTPS is enabled.
3. **Azure Database for PostgreSQL Flexible Server:** Managed PostgreSQL service that stores Mattermost data, with optional **same-zone** or **zone-redundant** high availability and configurable backup retention and geo-redundancy.
4. **Azure Files (NFS) with private endpoint:** Shared storage for Mattermost data (uploads, plugins, and other shared files), reachable only from inside the virtual network.
5. **Virtual network and subnets:** A dedicated VNet for the deployment, with separate subnets for the Application Gateway, the Mattermost application instances, the database, and the storage private endpoint.
6. **NAT gateway:** Provides controlled outbound internet access for the application instances (for example, to fetch packages and the Mattermost configuration scripts).
7. **Private DNS zones:** Used to resolve the PostgreSQL Flexible Server and Azure Files private endpoint from inside the VNet.

> Step 1: Select a plan and start creation

Reviewer: For each of the steps, could we add a screenshot of the wizard filled in with data?
Example the Azure tab: http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/8985/deployment-guide/server/deploy-kubernetes.html

Author: Yes, that's a good idea. Added in https://github.com/mattermost/docs/pull/8985/commits/892bf70b9e5ae6b6412a25e7b633667be4dee939

2. **Resource group:** Select an existing resource group or create a new one. A **new** resource group keeps the production stack easy to manage and clean up.
3. **Region:** Select the deployment region. Choose a region that supports the PostgreSQL Flexible Server SKUs you plan to use, and that has the **availability zones** you need if you intend to enable PostgreSQL high availability.
4. **Resource name prefix:** Short prefix (2–12 lowercase letters, digits, or hyphens) used to name the Azure resources created by the template (for example, ``mm-prod``).

Step 3: Application
-------------------

On the **Application** tab, configure how the Mattermost application is sized and how administrators access the VMs.

1. **Scaling tier:** Pick the user scale you are planning for. The wizard uses this to suggest sensible defaults for the **VMSS instance count**, **VM size**, and **PostgreSQL SKU**, and to determine whether **read replicas** are created for the database. Suggested tiers:

   * **Up to 200 users:** 1 node, ``Standard_F2s_v2``.
   * **Up to 2,000 users:** 2 nodes, ``Standard_F2s_v2``.
   * **Up to 15,000 users:** 2 nodes, ``Standard_F4s_v2``.

>    * **Up to 30,000 users:** 2 nodes, ``Standard_F8s_v2``.

Reviewer: To avoid duplication and maintainance of reference architectures:

```suggestion
1. **Scaling tier:** Pick the user scale you are planning for. The wizard uses this to suggest sensible defaults for the **VMSS instance count**, **VM size**, and **PostgreSQL SKU**, and to determine whether **read replicas** are created for the database. "The dropdown shows the recommended node count and VM SKU for each tier. For the underlying sizing rationale, see ::ref:`available reference architectures <administration-guide/scale/scaling-for-enterprise:available reference architectures>`, 
```

Author: https://github.com/mattermost/docs/pull/8985/commits/bde57aad53029d2f937c06bf9e509e9fb4ab19f5

> 2. **VMSS instance count:** Number of Mattermost application nodes (1–20). Defaults to the suggested value for your scaling tier; adjust if needed.

Reviewer: ARM script allows till 20 but UI dropdown is till 5. With the supported tiers, we wouldn't need more than 2:

```suggestion
2. **VMSS instance count:** Number of Mattermost application nodes (1–5). Defaults to the suggested value for your scaling tier; adjust if needed.
```

Author: https://github.com/mattermost/docs/pull/8985/commits/bde57aad53029d2f937c06bf9e509e9fb4ab19f5

3. **Enterprise license file:** Required when the instance count is greater than 1 (multi-node high-availability clustering). Upload your Mattermost Enterprise license file. Single-node deployments don't require a license.
4. **VM SKU:** Size of each VMSS instance. The wizard surfaces recommended Linux sizes first; you can select any supported Linux size available in your region.
5. **Admin username:** Linux administrator account used to sign in to the VM instances.
6. **Authentication type:** Choose **Password** or **SSH public key** for Linux sign-in. **SSH public key** is recommended for production deployments.
7. **Mattermost Version:** The Mattermost version to install (for example, ``11.6.0``). See the :doc:`Mattermost release policy </product-overview/release-policy>` for supported versions.

> Step 4: Database

Reviewer: Would it make sense to add somewhere https://learn.microsoft.com/en-us/azure/postgresql/overview#azure-regions as an easy way for customers to see availability of HA options?

Author: Yes 👍

Author: https://github.com/mattermost/docs/pull/8985/commits/c3abdc6b22fa4c98c9c8f564adeddca8d17108b4

1. **PostgreSQL compute SKU:** Memory-optimized Flexible Server SKU. Defaults to the SKU recommended for your scaling tier; pick a SKU that is supported in your region. Available choices include ``Standard_E2ds_v5``, ``Standard_E4ds_v5``, and ``Standard_E8ds_v5``.
2. **PostgreSQL high availability:** Choose how PostgreSQL is made resilient:

   * **Disabled:** Single primary instance. Suitable for small deployments and lower-cost evaluations.
   * **Same zone:** Primary and standby in the same availability zone.
   * **Zone redundant:** Primary and standby in different availability zones; recommended for production HA.

3. **Primary availability zone** and **standby availability zone:** Shown when HA is enabled. For zone-redundant HA, the primary and standby zones must be different. Pick zones supported in your region for the PostgreSQL SKU you chose.
4. **PostgreSQL admin username and password:** Administrator credentials for the Flexible Server. Avoid reserved names such as ``azure_superuser``. The password must meet `Azure Flexible Server password complexity requirements <https://learn.microsoft.com/azure/postgresql/flexible-server/concepts-security#password>`_.

> 5. **NFS share size (GiB):** Size of the Azure Files (NFS) share used by Mattermost for shared application data. Minimum 100 GiB.

Reviewer: We're missing the Mattermost database user password field right before NFS

Author: good catch, thanks

Author: https://github.com/mattermost/docs/pull/8985/commits/c3abdc6b22fa4c98c9c8f564adeddca8d17108b4

> 6. **Geo-redundant backup:** Replicates PostgreSQL backups to Azure's paired region for cross-region disaster recovery. Recommended for production. **This setting cannot be changed after deployment.**

Reviewer: Suggestion to remove the production recommendation, as this is not Mattermost specific and ties more to the DR requirements of each organization.
```suggestion
6. **Geo-redundant backup:** Replicates PostgreSQL backups to Azure's paired region for cross-region disaster recovery. **This setting cannot be changed after deployment.**
```

Author: https://github.com/mattermost/docs/pull/8985/commits/bde57aad53029d2f937c06bf9e509e9fb4ab19f5

.. note::

  Backup retention for PostgreSQL is set to 35 days at deployment time. You can change retention from the Azure portal after the deployment completes.

Step 5: Networking
------------------

On the **Networking** tab, configure the virtual network and how Mattermost is exposed to users.

1. **VNet address space (CIDR):** The address space for the virtual network created by the template. Defaults to ``10.0.0.0/22``. Pick a range that does not overlap with other VNets you plan to peer with.

> 2. **Public IP address:** Typically **create new**. You must set a **DNS prefix** (label) that is **globally unique** across Azure public cloud; it forms part of your URL and allows public access to your workspace. If the label is already taken, deployment will fail.

Reviewer: Suggestion to update field to match the wizard. There's no option in the Wizard to create a new one right?

```suggestion
2. **Public IP DNS label:**  You must set a **DNS prefix** (label) that is **globally unique** across Azure public cloud; it forms part of your URL and allows public access to your workspace. If the label is already taken, deployment will fail.
```

Author: https://github.com/mattermost/docs/pull/8985/commits/bde57aad53029d2f937c06bf9e509e9fb4ab19f5

3. **Enable HTTPS on Application Gateway:** When enabled, the Application Gateway terminates TLS using your PFX certificate and HTTP traffic is redirected to HTTPS. When disabled, Mattermost is served over HTTP on the Azure-assigned hostname (suitable for testing, **not recommended for production**).
4. **PFX certificate** and **PFX password:** Shown when HTTPS is enabled. Upload your PKCS#12 bundle and provide the password used to protect it.
5. **Custom domain (FQDN):** Shown when HTTPS is enabled. The public hostname Mattermost will use (for example, ``mattermost.example.com``). After deployment, point this hostname to the Azure-assigned DNS name of the Application Gateway public IP (typically using a **CNAME** record for subdomains).

Step 6: Review and create
-------------------------

> Review your settings, accept any **Marketplace** terms if prompted, then select **Create**. Provisioning the full stack typically takes longer than a single-VM deployment because of the database, storage, and Application Gateway resources. First-boot configuration on the Mattermost instances may take additional time before the application is reachable.

Reviewer: Suggestion swapping the current paragraph for a concrete timing range plus a quick "Azure Succeeded doesn't mean the URL is ready yet" note would give readers a clear expectation at this point in the flow.

```suggestion
Review your settings, accept any **Marketplace** terms if prompted, then select **Create**. Provisioning typically takes 20–50 minutes, depending on your PostgreSQL configuration. Allow a few additional minutes after Azure reports the deployment as Succeeded before the application is reachable.
```

Author: https://github.com/mattermost/docs/pull/8985/commits/bde57aad53029d2f937c06bf9e509e9fb4ab19f5

2. **Public IP address:** Typically **create new**. You must set a **DNS prefix** (label) that is **globally unique** across Azure public cloud; it forms part of your URL and allows public access to your workspace. If the label is already taken, deployment will fail.
3. **Enable HTTPS on Application Gateway:** When enabled, the Application Gateway terminates TLS using your PFX certificate and HTTP traffic is redirected to HTTPS. When disabled, Mattermost is served over HTTP on the Azure-assigned hostname (suitable for testing, **not recommended for production**).
4. **PFX certificate** and **PFX password:** Shown when HTTPS is enabled. Upload your PKCS#12 bundle and provide the password used to protect it.
5. **Custom domain (FQDN):** Shown when HTTPS is enabled. The public hostname Mattermost will use (for example, ``mattermost.example.com``). After deployment, you must create a **CNAME** record pointing this hostname to the Azure-assigned DNS name of the Application Gateway public IP.

> 

Author: https://github.com/mattermost/docs/pull/8985/commits/2c08d49bae6dc1c64c64f862b2d9c325bbafc7c2

> 

Author: https://github.com/mattermost/docs/pull/8985/commits/aa786ed9508e0c29bebf506a2459d65309fb38e1

Open the **Mattermost - Azure-Native (VM-based)** offer in the Azure Marketplace, click **Get it now**, select the plan, and then click **Create** to open the deployment wizard.

Step 2: Basics
~~~~~~~~~~~~~~

On the **Basics** tab, configure the following:

1. **Subscription:** Choose the Azure subscription where the deployment will live.
2. **Resource group:** Select an existing resource group or create a new one. A **new** resource group keeps the production stack easy to manage and clean up.
3. **Region:** Select the deployment region. Choose a region that supports the PostgreSQL Flexible Server SKUs you plan to use, and that has the **availability zones** you need if you intend to enable PostgreSQL high availability.
4. **Resource name prefix:** Short prefix (2–12 lowercase letters, digits, or hyphens) used to name the Azure resources created by the template (for example, ``mm-prod``).

  .. image:: /_static/images/azure/azure-native-basics.png
    :alt: An example of the Azure Native Basics setup screen.

Step 3: Application
~~~~~~~~~~~~~~~~~~~

On the **Application** tab, configure how the Mattermost application is sized and how administrators access the VMs.

> 1. **Scaling tier:** Pick the user scale you are planning for. The wizard uses this to suggest sensible defaults for the **VMSS instance count**, **VM size**, and **PostgreSQL SKU**, and to determine whether **read replicas** are created for the database. "The dropdown shows the recommended node count and VM SKU for each tier. For the underlying sizing rationale, see ::ref:`available reference architectures <administration-guide/scale/scaling-for-enterprise:available reference architectures>`.

Reviewer: ```suggestion
1. **Scaling tier:** Pick the user scale you are planning for. The wizard uses this to suggest sensible defaults for the **VMSS instance count**, **VM size**, and **PostgreSQL SKU**, and to determine whether **read replicas** are created for the database. "The dropdown shows the recommended node count and VM SKU for each tier. For the underlying sizing rationale, see :ref:`available reference architectures <administration-guide/scale/scaling-for-enterprise:available reference architectures>`.
```

Author: https://github.com/mattermost/docs/pull/8985/commits/7604485d44911a633061a78a5e73586b175fa4f9

2. **VMSS instance count:** Number of Mattermost application nodes (1–5). Defaults to the suggested value for your scaling tier; adjust if needed.
3. **Enterprise license file:** Required when the instance count is greater than 1 (multi-node high-availability clustering). Upload your Mattermost Enterprise license file. The license is optional for single-node deployments.
4. **VM SKU:** Size of each VMSS instance. The wizard surfaces recommended Linux sizes first; you can select any supported Linux size available in your region.
5. **Admin username:** Linux administrator account used to sign in to the VM instances.
6. **Authentication type:** Choose **Password** or **SSH public key** for Linux sign-in. **SSH public key** is recommended for production deployments.

> 7. **Mattermost Version:** The Mattermost version to install (for example, ``11.6.0``). See the :doc:`Mattermost release policy </product-overview/release-policy>` for supported versions.

Reviewer: nit: let's add a version we have available for them, e.g. 1.6.2

Author: https://github.com/mattermost/docs/pull/8985/commits/7604485d44911a633061a78a5e73586b175fa4f9

.. image:: /_static/images/azure/azure-native-application.png
  :alt: An example of the Azure Native Application setup screen.

Step 4: Database
~~~~~~~~~~~~~~~~

On the **Database** tab, configure the managed PostgreSQL service and the shared NFS file share.

> 1. **PostgreSQL compute SKU:** Memory-optimized Flexible Server SKU. Defaults to the SKU recommended for your scaling tier; pick a SKU that is supported in your region. Available choices include ``Standard_E2ds_v5``, ``Standard_E4ds_v5``, and ``Standard_E8ds_v5``.

Reviewer: I'd either change the `include` to `are` since these are the only available ones, or I'd remove this sentence completely.

Author: https://github.com/mattermost/docs/pull/8985/commits/7604485d44911a633061a78a5e73586b175fa4f9

2. **PostgreSQL high availability:** Choose how PostgreSQL is made resilient:

>    * **Disabled:** Single primary instance. Suitable for small deployments and lower-cost evaluations.

Reviewer: I'd remove the lower-cost evaluations as we'll be redirecting them to the quickstart offering for evaluations

Author: https://github.com/mattermost/docs/pull/8985/commits/7604485d44911a633061a78a5e73586b175fa4f9

**After deployment completes:**

* In the Azure portal, open the **resource group** you used.
* Open the **public IP resource attached to your Application Gateway** (often named ``<prefix>-ag-pip``). Under **Essentials**, note the **DNS name** (Azure-assigned FQDN).
* If you deployed **without HTTPS**, your site URL is:

  ``http://<dns-label>.<region>.cloudapp.azure.com``

* If you deployed **with HTTPS**, create a DNS record that points your **custom domain** to the Azure-assigned DNS name of the Application Gateway public IP (typically **CNAME** for subdomains; use your provider's apex-compatible record type if using a root domain). Once DNS propagates, your site URL is:

  ``https://<your-custom-domain>``

Step 7: Open Mattermost and create your administrator
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

In a browser, go to your Mattermost URL. Mattermost will prompt you to **create the first user**, which becomes the **System Administrator**. That account is different from the **Linux** username and password (or SSH key) you configured in Step 3.

> Congratulations! You've successfully deployed a production-style Mattermost stack on Azure.

Reviewer: I'd remove this section, as this is not related to the deployment.

Author: https://github.com/mattermost/docs/pull/8985/commits/7604485d44911a633061a78a5e73586b175fa4f9

:orphan:
:nosearch:

> Mattermost is published as an **Azure Marketplace** solution that provisions a production-style Mattermost stack in your own subscription using Azure-managed services for compute, database, shared file storage, and load balancing. This option is intended for organizations that want a resilient Mattermost deployment they can scale and operate within their existing Azure tenant.

Reviewer: There are 3 places in the doc where we mention `production-style`. I wonder if saying `production-ready` is better wording. Thoughts ?

Author: Agreed and is also aligned with our public facing description in the offer 👍

Author: https://github.com/mattermost/docs/pull/8985/commits/e412b90a6e8a5e77b91642c438a55d71da2b675d

* Familiarity with the **Azure portal** (resource groups, networking, virtual machines, and Application Gateway).
* For multi-node deployments (more than one application instance), a valid **Mattermost Enterprise license** file. The license is optional for single-node deployments.
* If you plan to enable **HTTPS** at deployment time:

  * A **PFX (PKCS#12)** certificate bundle and its password.
  * A **custom domain (FQDN)** that you will point to the Application Gateway after deployment.

What gets deployed
~~~~~~~~~~~~~~~~~~

The template provisions a self-contained, production-style stack in the resource group you select, including:

1. **Virtual Machine Scale Set (Linux):** Runs Mattermost on Ubuntu Pro 24.04 LTS images published by Mattermost. Instance count and VM size are driven by the scaling tier you choose.
2. **Application Gateway with public IP:** Acts as the entry point for Mattermost traffic, with optional **HTTPS listener** (using your PFX certificate) and HTTP-to-HTTPS redirection when HTTPS is enabled.

> 3. **Azure Database for PostgreSQL Flexible Server:** Managed PostgreSQL service that stores Mattermost data, with optional **same-zone** or **zone-redundant** high availability and configurable backup retention and geo-redundancy.

Reviewer: I missed this in earlier reviews, should this be? 
```suggestion
3. **Azure Database for PostgreSQL Flexible Server:** Managed PostgreSQL service that stores Mattermost data, with optional **same-zone** or **zone-redundant** high availability, backup retention and geo-redundancy.
```
Backup retention isn't configurable (set at 35 days) and geo-redundancy is optional

Author: Ah yes good catch 👍

Author: https://github.com/mattermost/docs/pull/8985/commits/e412b90a6e8a5e77b91642c438a55d71da2b675d

Open the **Mattermost - Azure-Native (VM-based)** offer in the Azure Marketplace, click **Get it now**, select the plan, and then click **Create** to open the deployment wizard.

Step 2: Basics
~~~~~~~~~~~~~~

On the **Basics** tab, configure the following:

1. **Subscription:** Choose the Azure subscription where the deployment will live.
2. **Resource group:** Select an existing resource group or create a new one. A **new** resource group keeps the production stack easy to manage and clean up.
3. **Region:** Select the deployment region. Choose a region that supports the PostgreSQL Flexible Server SKUs you plan to use, and that has the **availability zones** you need if you intend to enable PostgreSQL high availability.
4. **Resource name prefix:** Short prefix (2–12 lowercase letters, digits, or hyphens) used to name the Azure resources created by the template (for example, ``mm-prod``).

  .. image:: /_static/images/azure/azure-native-basics.png
    :alt: An example of the Azure Native Basics setup screen.

Step 3: Application
~~~~~~~~~~~~~~~~~~~

On the **Application** tab, configure how the Mattermost application is sized and how administrators access the VMs.

> 1. **Scaling tier:** Pick the user scale you are planning for. The wizard uses this to suggest sensible defaults for the **VMSS instance count**, **VM size**, and **PostgreSQL SKU**, and to determine whether **read replicas** are created for the database. "The dropdown shows the recommended node count and VM SKU for each tier. For the underlying sizing rationale, see :ref:`available reference architectures <administration-guide/scale/scaling-for-enterprise:available reference architectures>`.

Reviewer: nit:

```suggestion
1. **Scaling tier:** Pick the user scale you are planning for. The wizard uses this to suggest sensible defaults for the **VMSS instance count**, **VM size**, and **PostgreSQL SKU**, and to determine whether **read replicas** are created for the database. The dropdown shows the recommended node count and VM SKU for each tier. For the underlying sizing rationale, see :ref:`available reference architectures <administration-guide/scale/scaling-for-enterprise:available reference architectures>`.
```

Author: https://github.com/mattermost/docs/pull/8985/commits/e412b90a6e8a5e77b91642c438a55d71da2b675d

2. **VMSS instance count:** Number of Mattermost application nodes (1–5). Defaults to the suggested value for your scaling tier; adjust if needed.
3. **Enterprise license file:** Required when the instance count is greater than 1 (multi-node high-availability clustering). Upload your Mattermost Enterprise license file. The license is optional for single-node deployments.
4. **VM SKU:** Size of each VMSS instance. The wizard surfaces recommended Linux sizes first; you can select any supported Linux size available in your region.
5. **Admin username:** Linux administrator account used to sign in to the VM instances.
6. **Authentication type:** Choose **Password** or **SSH public key** for Linux sign-in. **SSH public key** is recommended for production deployments.

> 7. **Mattermost Version:** The Mattermost version to install (for example, ``11.6.2``). See the :doc:`Mattermost release policy </product-overview/release-policy>` for supported versions.

Reviewer: nit for consistancy
```suggestion
7. **Mattermost version:** The Mattermost version to install (for example, ``11.6.2``). See the :doc:`Mattermost release policy </product-overview/release-policy>` for supported versions.
```

Author: https://github.com/mattermost/docs/pull/8985/commits/e412b90a6e8a5e77b91642c438a55d71da2b675d

.. image:: /_static/images/azure/azure-native-application.png
  :alt: An example of the Azure Native Application setup screen.

Step 4: Database
~~~~~~~~~~~~~~~~

On the **Database** tab, configure the managed PostgreSQL service and the shared NFS file share.

1. **PostgreSQL compute SKU:** Memory-optimized Flexible Server SKU. Defaults to the SKU recommended for your scaling tier; pick a SKU that is supported in your region. Available choices are ``Standard_E2ds_v5``, ``Standard_E4ds_v5``, and ``Standard_E8ds_v5``.
2. **PostgreSQL high availability:** Choose how PostgreSQL is made resilient:

   * **Disabled:** Single primary instance. Suitable for small deployments.
   * **Same zone:** Primary and standby in the same availability zone.

>    * **Zone redundant:** Primary and standby in different availability zones; recommended for production HA.

Reviewer: Missed this earlier. Suggestion to remove recommendation for 2 reasons:
1. Zone redundant is not available in many Azure regions
2. Same zone vs Zone redundant is a DR decision, regardless of Mattermost. 

```suggestion
   * **Zone redundant:** Primary and standby in different availability zones.
```

Author: https://github.com/mattermost/docs/pull/8985/commits/e412b90a6e8a5e77b91642c438a55d71da2b675d

Before you upgrade
^^^^^^^^^^^^^^^^^^

* Review the :doc:`important upgrade notes </administration-guide/upgrade/important-upgrade-notes>`, and :doc:`prepare to upgrade </administration-guide/upgrade/prepare-to-upgrade-mattermost>` for the target version.
* Confirm **PostgreSQL** backups are current and that you can restore if needed.
* Pick a target version from the :doc:`Mattermost release policy </product-overview/release-policy>`.

Upgrade steps
^^^^^^^^^^^^^

1. **Update the VMSS** to the new image version:

   .. code-block:: sh

     az vmss update \
       --resource-group <resource-group> \
       --name <resource-name-prefix>-vmss \
       --set virtualMachineProfile.storageProfile.imageReference.version=<new-version>

   The VMSS uses a **rolling** upgrade policy and an application health probe on ``/api/v4/system/ping``, so instances are replaced in batches while the Application Gateway keeps serving traffic to healthy nodes.

2. **Monitor the rollout** in the Azure portal: open your VMSS → **Instances** (or **Rolling upgrades**) and wait until all instances are healthy and running the new model.

> 3. **Verify Mattermost**: open your site, confirm users can sign in, and check **System Console → About** for the expected server version.

Reviewer: "confirm users can sign in" could cause uncertainty to customers (is it possible that my users won't be able to sign in?). Suggestion:
```suggestion
3. **Verify Mattermost**: Verify Mattermost uses the new version via **System Console → About**.
```

Author: https://github.com/mattermost/docs/pull/8985/commits/e412b90a6e8a5e77b91642c438a55d71da2b675d
